sormas-backend/src/main/java/de/symeda/sormas/backend/location/LocationFacadeEjb.java

```diff
@@ -172,13 +172,26 @@ public boolean areDifferentLocation(LocationDto firstAddress, LocationDto second
 			return true;
 		}
 
+		if (Boolean.TRUE.equals(!DataHelper.equal(firstAddress.getFacilityType(), secondAddress.getFacilityType()))) {
+			return true;
+		}
+
 		if (firstAddress.getFacility() != null) {
 			Facility firstAddressFacilityDto = facilityService.getByUuid(firstAddress.getFacility().getUuid());
 			if (secondAddress.getCommunity() != null
 				&& firstAddressFacilityDto.getCommunity() != null
 				&& !firstAddressFacilityDto.getCommunity().getUuid().equals(secondAddress.getCommunity().getUuid())) {
```

Comment:
please use `DataHelper.isSame(firstAddressFacilityDto.getCommunity(), secondAddress.getCommunity()) ` instead of comparing the uuids

Reply:
changed